hana/ha_cluster.sls

```diff
@@ -78,7 +98,18 @@ sudoers_edit_{{ sap_instance }}:
     - force: true
     - require:
       - sudoers_check_{{ sap_instance }}
-      - stop_hana_{{ sap_instance }}
+
+# Stop SAP Hana
```

Comment:
Did you check is this posterior `stop` works as expected and doesn't affect at any point the HANA processes?
In the documentation the stop is done before all of these updates (and this is how it was implemented).

If doing it later works, I'm fine with it, but I would put a more explicit comment here. Like `# Stop the HANA after applying the Cluster hooks. In the official guide this is done before the changes, but in order to make things easier, it is moved at the end`

Reply:
@arbulu89 Both parameters that we set (system_replication.operation_mode and ha_dr_provider) need a restart of the HANA if touched (https://help.sap.com/viewer/009e68bc5f3c440cb31823a3ec4bb95b/2.0.05/en-US/514ab38a2e574c85a70ebba80ff16d99.html).
I tried it out several times and also know from my past experience, that changing the global.ini is safe on a running system. You just need the restart to activate certain changes.
A conditional restart is of course only possible at the very end after changes are applied (or not).

Reply:
I also changed the way the `global.ini` file is processed to be idempotent, which makes me confident that this is an improvement.

Reply:
Ok, nice! Thank you for this nice explanation